kaolin/graphics/SoftRenderer.py

```diff
@@ -462,14 +495,19 @@ def look_at(vertices, eye, at=torch.FloatTensor([0, 0, 0]),
     def look(self, vertices, eye, direction=torch.FloatTensor([0, 1, 0]),
              up=None):
         r"""Apply the "look" transformation to the vertices.
+
+        Returns:
+            vertices (torch.Tensor): Input vertices transformed to the camera coordinate
+                frame (shape: :math:`B \times V \times 3`), where :math:`B` is the batchsize,
+                and :math:`V` is the number of vertices in the mesh.
         """
 
         import torch.nn.functional as F
 
         device = vertices.device
         direction = direction.to(device)
         if up is None:
-            up = torch.FloatTensor([0, 1, 0]).to(device)
+            up = torch.FloatTensor([0, 1, 0], device=device)
```

Comment:
For torch.FloatTensor the argument `device` actually doesn't work for anything else than `cpu`.
You wanna use torch.tensor (or the faster torch.cuda.FloatTensor if you want to enforce device=cuda).